src/builder/pod/mod.rs

```diff
@@ -367,7 +437,62 @@ impl PodBuilder {
         self
     }
 
+    /// Consumes the Builder and returns a constructed [`Pod`]
+    pub fn build(&self) -> OperatorResult<Pod> {
+        Ok(Pod {
+            metadata: match self.metadata {
+                None => return Err(Error::MissingObjectKey { key: "metadata" }),
+                Some(ref metadata) => metadata.clone(),
+            },
+            spec: Some(self.build_spec()),
+            status: self.status.clone(),
+        })
+    }
+
+    /// Returns a [`PodTemplateSpec`], usable for building a [`StatefulSet`](`k8s_openapi::api::apps::v1::StatefulSet`)
+    /// or [`Deployment`](`k8s_openapi::api::apps::v1::Deployment`)
+    pub fn build_template(&self) -> OperatorResult<PodTemplateSpec> {
+        if self.status.is_some() {
+            tracing::warn!("Tried building a PodTemplate for a PodBuilder with a status, the status will be ignored...");
+        }
+
+        Ok(PodTemplateSpec {
+            metadata: self.metadata.clone(),
+            spec: Some(self.build_spec()),
+        })
+    }
+
+    /// Checks if any of the containers is missing one resource requirement
+    /// type for `resource`. If so, [Error::MissingResourceRequirementType]
+    /// is returned, otherwise [`Ok(())`].
+    fn check_container_resource(
+        &self,
+        rr_type: ResourceRequirementsType,
+        resource: &str,
+    ) -> OperatorResult<()> {
+        for container in &self.containers {
+            container.check_resource_requirement(rr_type, resource)?
+        }
+
+        Ok(())
+    }
+
     fn build_spec(&self) -> PodSpec {
+        // We don't hard error here, because if we do, the StatefulSet (for
+        // example) doesn't show up at all. Instead users then need to comb
+        // through the logs to find the error. That's why we opted to just
+        // throw a warning which will get displayed in the Kubernetes
+        // status.
+
+        if let Err(err) = self.check_container_resource(ResourceRequirementsType::Limits, "cpu") {
```

Comment:
I would rather build the PodSpec, and than use
```
impl ResourceRequirementsExt for PodSpec {
    fn check_resource_requirement(
        &self,
        rr_type: ResourceRequirementsType,
        resource: &str,
    ) -> Result<(), ResourceRequirementsError> {
        for container in &self.containers {
            container.check_resource_requirement(rr_type, resource)?
        }

        Ok(())
    }
}
```

We also need to check this in `ClusterResource::add`, otherwise it was useless to extend the Trait ;)
```
        if let Some(pod_spec) = resource.pod_spec() {
            if let Err(err) =
                pod_spec.check_resource_requirement(ResourceRequirementsType::Limits, "cpu")
            {
                warn!("{}", err)
            }
            if let Err(err) =
                pod_spec.check_resource_requirement(ResourceRequirementsType::Limits, "memory")
            {
                warn!("{}", err)
            }
        }
```

Reply:
Added in 597e751